xblock/test/test_parsing.py

```diff
@@ -450,6 +450,32 @@ def test_nested_roundtrip(self):
         self.assertEqual(parsed.sequence, expected_seq)
         self.assertEqual(parsed.dictionary, expected_dict)
 
+    @XBlock.register_temp_plugin(Leaf)
+    def test_block_fields_if_none(self):
+        """
+        Verify that if block fields (setting fields) are of NoneType
+        that its does break rather set empty string to node value.
+        """
+        block = self.parse_xml_to_block("<leaf/>")
+
+        # example an setting field git_url  fo block
+        git_url = String(display_name="GIT URL", scope=Scope.settings, default=None)
+        git_url.__name__ = "git_url"
+
+        # set as mark as dirty so that we can set the value on node
+        # pylint: disable=protected-access
+        git_url._mark_dirty(block, git_url)
+
+        # add the setting field to git_url to block fields
+        block.fields['git_url'] = git_url
+
+        # pylint: disable=protected-access
+        block._field_data_cache['git_url'] = None
```

Comment:
I don't understand what the previous three lines do. It seems like the code is unconventionally setting `git_url` to None. Why don't conventional means work? This code will add mystery - and no one wants to touch mysterious code after it's merged. The comments for this code should be more verbose.